### DISCUSSION TITLE
add removal tool in vectorizer

### DISCUSSION THREAD

**Discussion by slvamegestic** [clientIO/joint#1984] [Ideas]
Hi I am using jointjs 3.6.5,

`  var line = V("rect", {
        x: 20,
        y: 20
        width: 100,
        height: 100,
        "stroke-width": 3,
        fill: "transparent",
        stroke: "#142f5a",
      });
      V(this.bgPaper.viewport).append(line);`

<img width="595" alt="Screenshot 2023-01-12 at 5 39 58 PM" src="https://user-images.githubusercontent.com/35224544/212063038-935e7f8c-d604-4b54-9e79-a07d3eea4b31.png">

The above code draws a rectangle as a line(the image shows multiple rectangle lines in black). When hovering over the lines, I want a removal tool to appear. How can I do that ?

**Comment by kumilingus** [comment]
If you choose to use low-level APIs (*Vectorizer or native SVG*) to draw objects, you can't expect high-level classes (*elementTools, linkTools*) to work with them (they only work with *ElementView* and *LinkView*).

You need to attach event listeners using DOM API and render the button using Vectorizer or DOM API.

**Reply by slvamegestic** [reply]
i have done that using the below code
`
  $('svg').click(function(event){
       $(event.target).remove()
}
`
It works fine and the SVG is removed when I click on an SVG rectangle. I would appreciate it if you could suggest a way to render the x button while hovering over the SVG.

**Reply by kumilingus** [reply]
The same way you have rendered a rectangle. But this time you use `<path/>` element with `<circle/>` element as background. And you need to append the elements to the DOM. You can listen to `mousenter` event.

**Reply by slvamegestic** [reply]
My best efforts were made. But the `mouseenter` event is not even triggered while hovering over my SVG rectangle. Can you please explain more about this with a snippet/fiddle?

**Reply by kumilingus** [reply]
Please try to create the fiddle and we'll look at what is wrong.

*Note that this is not a question for JointJS, but a general question about working with DOM.*

**Reply by slvamegestic** [reply]
Noted on the above @kumilingus 
  Please see the simple fiddle I have done regarding removing a Vectorizer line here [click here](https://jsfiddle.net/selvakkumar/qm7hx4n3/4/).
 I have removed the rect line when clicking it, but I need to append the x icon when hovering over the line. Once the X has been clicked, the line needs to be removed.

Thanks for you time to read this

**Reply by kumilingus** [reply]
Please see updated [jsfiddle](https://jsfiddle.net/kumilingus/mq9wr1o6/8/).

**Reply by slvamegestic** [reply]
That's great 👍. My gratitude goes out to you @kumilingus.
